arch/x86/kvm/pkvm/pkvm.c

```diff
@@ -508,6 +508,55 @@ static int pkvm_vcpu_load(int vm_handle, int vcpu_handle)
 	return ret;
 }
 
+static int pkvm_vcpu_put(int vm_handle, int vcpu_handle)
+{
+	struct pkvm_vcpu *pkvm_vcpu = pkvm_get_vcpu(vm_handle, vcpu_handle);
+	int cpu = raw_smp_processor_id(), loaded_cpu, ret = 0;
+	struct kvm_vcpu *vcpu;
+
+	if (!pkvm_vcpu)
+		return -EINVAL;
+
+	vcpu = &pkvm_vcpu->vcpu;
+	loaded_cpu = READ_ONCE(vcpu->cpu);
+	if (loaded_cpu == cpu) {
+		/*
+		 * The current active vCPU is the host vCPU. Switch to the guest
+		 * vCPU in case vcpu_put operation requires.
```

Comment:
Question: does vmx_vcpu_put() actually require that?

Looks like it doesn't (looks like basically all it does is VMCLEAR). Also I've just quickly tried commenting out this vcpu_load, and VMs still seem to be working without problems.

So if it is not required, maybe let's not do it, to keep the code simpler?

Reply:
Ok, seems both ways have pros and cons (for example, removing this vcpu_load would avoid redundant VMCLEAR, but OTOH, keeping this vmcs_load may be desirable as KVM might have an implicit assumption that vcpu_put is always paired with a prior vcpu_load of the same vCPU). So it's probably ok to keep it.